sdk/keyvault/keyvault-keys/review/keyvault-keys.api.md

```diff
@@ -57,7 +87,11 @@ export interface CreateRsaKeyOptions extends CreateKeyOptions {
 export class CryptographyClient {
     constructor(key: string | KeyVaultKey, credential: TokenCredential, pipelineOptions?: CryptographyClientOptions);
     constructor(key: JsonWebKey);
+    decrypt(decryptParameters: DecryptParameters, options?: DecryptOptions): Promise<DecryptResult>;
+    // @deprecated
```

Comment:
These shouldn't be deprecated. You just added overloads. Besides, the parameters above don't support the existing RSA algorithms, which the `encrypt` and `decrypt` methods you're deprecating still do/would.

Reply:
I see you defined them below and OR'd them with the type union. Still, these methods shouldn't be deprecated. If that causes builds to start failing (even warning, which people might force to errors) existing builds would break without modifications. That's why we don't use the `[Obsolete]` attribute in .NET>

Reply:
In our case @deprecated is generally safe to use (confirmed with the team). For this particular change my reasoning was:

- The existing methods will not work for any AES algorithms due to lack of needed parameters
- The new overloads are a superset of the existing methods. They _do_ support RSA algorithms and AES algorithms (everything the existing methods support today and more)

It felt like the existing methods are deprecated as they do not support all the encryption parameters the service supports. They'll continue to be maintained for now but maybe we want to signal that the new methods should be used instead?

@bterlson what do you think? I opened a PR to undeprecate these and I'll be happy to defer to you since it's an API change.

Reply:
commented in the other PR